packages/react-components/src/components/MessageThread.test.tsx

```diff
@@ -225,3 +227,292 @@ describe('Message should display inline image correctly', () => {
+  test('Edit Message with @ will show MentionPopover and mentions in edited message', async () => {
+    const user1Id = 'user1';
+    const user2Id = 'user2';
+
+    const user1Name = 'Kat Larsson';
+
+    const messages: ChatMessage[] = [
+      {
+        messageType: 'chat',
+        senderId: user2Id,
+        senderDisplayName: 'Robert Tolbert',
+        messageId: generateGUID(),
+        content: 'Absolutely! What seems to be the problem?',
+        createdOn: new Date('2019-04-13T00:00:00.000+08:11'),
+        mine: true,
+        attached: false,
+        contentType: 'html'
+      }
+    ];
+
+    const expectedOnUpdateMessageCount = 1;
+    let onUpdateMessageCount = 0;
+    const onUpdateMessageCallback = (messageId, content): Promise<void> => {
+      const msgIdx = messages.findIndex((m) => m.messageId === messageId);
+      const message = messages[msgIdx];
+      message.content = content;
+      message.editedOn = new Date(Date.now());
+      messages[msgIdx] = message;
+      onUpdateMessageCount++;
+      return Promise.resolve();
+    };
+
+    const onQueryUpdated = async (query: string): Promise<Mention[]> => {
+      return Promise.resolve(
+        [
+          {
+            id: user1Id,
+            displayText: user1Name
+          },
+          {
+            id: 'everyone',
+            displayText: 'Everyone'
+          }
+        ].filter((suggestion) => suggestion.displayText.toLocaleLowerCase().startsWith(query.toLocaleLowerCase()))
+      );
+    };
+
+    const { container, rerender } = render(
+      <MessageThread
+        userId={user2Id}
+        messages={messages}
+        onUpdateMessage={onUpdateMessageCallback}
+        mentionOptions={{
+          displayOptions: {
+            onRenderMention: (mention, defaultOnMentionRender) => {
+              return <span key={generateGUID()}>{defaultOnMentionRender(mention)}</span>;
+            }
+          },
+          lookupOptions: {
+            onQueryUpdated: onQueryUpdated
+          }
+        }}
+      />
+    );
+
+    // Find message bubble does not contain mention yet
+    const messageBubble = container.querySelector('[data-ui-id="chat-composite-message"]');
+    if (!messageBubble) {
+      fail('it should not reach here');
+    }
+    expect(messageBubble.innerHTML).not.toContain(user1Name);
+    expect(messageBubble.innerHTML).not.toContain(MSFT_MENTION);
+
+    // Click on ... button to trigger context menu
+    const menuButton = container.querySelector('[data-ui-id="chat-composite-message-action-icon"]');
+    if (!menuButton) {
+      fail('it should not reach here');
+    }
+    fireEvent.click(menuButton);
+
+    // Click on Edit ContextMenuItem
+    const editButton = await screen.findByText('Edit');
+    fireEvent.click(editButton);
+
+    // Type ' @' in edit box to show mentions popover menu
+    const editBox = await screen.getByPlaceholderText('Edit your message');
+    await waitFor(async () => {
+      userEvent.keyboard(' @');

```

Comment:
Should 'await' be added here to wait while the whole text is added to the input box?